test/parallel/test-http-early-hints.js

```diff
@@ -0,0 +1,161 @@
+'use strict';
+const common = require('../common');
+const assert = require('node:assert');
+const http = require('node:http');
+const debug = require('node:util').debuglog('test');
+
+const testResBody = 'response content\n';
+
+{
+  const server = http.createServer(common.mustCall((req, res) => {
+    debug('Server sending early hints...');
+    res.writeEarlyHints('</styles.css>; rel=preload; as=style');
+
+    debug('Server sending full response...');
+    res.end(testResBody);
+  }));
+
+  server.listen(0, common.mustCall(() => {
+    const req = http.request({
+      port: server.address().port, path: '/'
+    });
+
+    debug('Client sending request...');
+
+    req.on('information', common.mustCall((res) => {
+      assert.strictEqual(res.headers.link, '</styles.css>; rel=preload; as=style');
+    }));
+
+    req.on('response', common.mustCall((res) => {
+      let body = '';
+
+      assert.strictEqual(res.statusCode, 200, `Final status code was ${res.statusCode}, not 200.`);
+
+      res.on('data', common.mustCall((chunk) => {
+        body += chunk;
+      }));
+
+      res.on('end', common.mustCall(() => {
+        debug('Got full response.');
+        assert.strictEqual(body, testResBody);
+        server.close();
+      }));
+    }));
+
+    req.end();
+  }));
+}
+
+{
+  const server = http.createServer(common.mustCall((req, res) => {
+    debug('Server sending early hints...');
+    res.writeEarlyHints([
+      '</styles.css>; rel=preload; as=style',
+      '</scripts.js>; rel=preload; as=script',
+    ]);
+
+    debug('Server sending full response...');
+    res.end(testResBody);
+  }));
+
+  server.listen(0, common.mustCall(() => {
+    const req = http.request({
+      port: server.address().port, path: '/'
+    });
+    debug('Client sending request...');
+
+    req.on('information', common.mustCall((res) => {
+      assert.strictEqual(res.headers.link, '</styles.css>; rel=preload; as=style, </scripts.js>; rel=preload; as=script');
+    }));
+
+    req.on('response', common.mustCall((res) => {
+      let body = '';
+
+      assert.strictEqual(res.statusCode, 200, `Final status code was ${res.statusCode}, not 200.`);
+
+      res.on('data', common.mustCall((chunk) => {
```

Comment:
I think we should probably avoid `common.mustCall()` on this one because we shouldn't assume the number of chunks we get and we're already checking the contents of `body` in the `'end'` event handler anyway.

Reply:
true, done! also did some other cleanup to keep the test case to the point